Simplenote/src/main/java/com/automattic/simplenote/NoteEditorFragment.java

```diff
@@ -887,15 +887,22 @@ public void onTextChanged(CharSequence charSequence, int start, int before, int
     }
 
     private void setTitleSpan(Editable editable) {
-        // Set the note title to be a larger size
-        // Remove any existing size spans
-        RelativeSizeSpan[] spans = editable.getSpans(0, editable.length(), RelativeSizeSpan.class);
-        for (RelativeSizeSpan span : spans) {
+        // Set the note title to be a larger size and bold style.
+        // NOTE: Remove all existing spans before applying spans or performance issues will occur.
+        for (RelativeSizeSpan span : editable.getSpans(0, editable.length(), RelativeSizeSpan.class)) {
             editable.removeSpan(span);
         }
+
+        for (StyleSpan span : editable.getSpans(0, editable.length(), StyleSpan.class)) {
+            editable.removeSpan(span);
+        }
```

Comment:
would it be appropriate to call `clearSpans()` here instead of iterating over all of them and removing them?
https://developer.android.com/reference/android/text/Editable#clearSpans()

```java
private void setTitleSpan(Editable editor) {
	// if we don't clear out the old spans then we will leak them
	// and our performance will suffer as the title gets longer
	// we will add back the appropriate ones below
	editable.clearSpans();

	int newLinePosition = getNoteContentString().indexOf("\n");
	…
}
```

Reply:
You would think so according to the documentation, but `Editable.clearSpans()` doesn't exactly work as expected.  All spans are removed, but it seems to disable applying future spans.  Also, it moves the cursor to the first position making text input very difficult.  Finally, there are significant performance issues even with a small number of characters.  See the animation below for illustration.

<a href="https://user-images.githubusercontent.com/3827611/75818482-2f476480-5d56-11ea-864b-4c1983d7e54f.gif"><img src="https://user-images.githubusercontent.com/3827611/75818482-2f476480-5d56-11ea-864b-4c1983d7e54f.gif" width="300"></a>

Reply:
Wow. That’s surprising. I guess it is really removing more spans that aren’t obviously there but which are important for proper functionality.